src/core/random.hpp

```diff
@@ -27,183 +27,46 @@
 */
 
 #include "utils.hpp"
+#include <random>
 
-/*----------------------------------------------------------*/
+void init_random(void);
+void init_random_seed(int seed);
 
-/* Stuff for Franks ran1-generator */
-/*@{*/
-#define IA 16807
-#define IM 2147483647
-#define AM (1.0/2147483647.)
-#define IQ 127773
-#define IR 2836
-#define NDIV ((double) (1+(2147483647-1)/NTAB_RANDOM))
-#define RNMX (1.0-1.2e-7)
-# define NTAB_RANDOM  32
+extern std::mt19937 generator;
```

Comment:
This construction defies the intent to inline the generation functions. (Other units see just the extern declaration, not the actual code => no inline)